java/test/jmri/ArchitectureTest.java

```diff
@@ -285,23 +285,62 @@ static public void tearDown() {
         .should().accessClassesThat().resideInAPackage("org.jdom2..");
 
     /**
-     * Confine purejavacomm to jmri.jmrix packages.
+     * Confine purejavacomm to existing uses. No additional ones permitted.
+     * Ideally these will all be migrated to jSerialComm eventually.
      */
     @ArchTest
-    public static final ArchRule checkPurejavacoomOutsideJmrix = noClasses()
-        .that().resideOutsideOfPackage("jmri.jmrix..").and()
+    public static final ArchRule checkPurejavacommUsage = noClasses()
+        .that()
+        //.resideOutsideOfPackage("jmri.jmrix..").and()
+        
         .doNotHaveFullyQualifiedName("apps.util.issuereporter.SystemInfo").and()
-        .doNotHaveFullyQualifiedName("jmri.jmrit.mailreport.ReportContext")
+        .doNotHaveFullyQualifiedName("jmri.jmrit.mailreport.ReportContext").and()
+
+        .doNotHaveFullyQualifiedName("jmri.jmrix.AbstractSerialConnectionConfig").and()
+        .doNotHaveFullyQualifiedName("jmri.jmrix.AbstractSerialPortController").and()
+        .doNotHaveFullyQualifiedName("jmri.jmrix.AbstractSerialPortController$1").and()
+
+        // non-typical systems that are not (yet) migrated
+        .doNotHaveFullyQualifiedName("jmri.jmrix.bachrus.SpeedoTrafficController").and()
+        .doNotHaveFullyQualifiedName("jmri.jmrix.bachrus.kpfserialdriver.SerialDriverAdapter").and()
+        .doNotHaveFullyQualifiedName("jmri.jmrix.bachrus.serialdriver.SerialDriverAdapter").and()
+        .doNotHaveFullyQualifiedName("jmri.jmrix.dcc4pc.Dcc4PcTrafficController").and()
+        .doNotHaveFullyQualifiedName("jmri.jmrix.dcc4pc.serialdriver.SerialDriverAdapter").and()
+        .doNotHaveFullyQualifiedName("jmri.jmrix.ieee802154.xbee.XBeeAdapter").and()
+        .doNotHaveFullyQualifiedName("jmri.jmrix.ieee802154.IEEE802154PortController").and()
+        .doNotHaveFullyQualifiedName("jmri.jmrix.ieee802154.serialdriver.SerialDriverAdapter").and()
```

Comment:
No need to change. But isn't NCE migrated to the new serial lib?

Reply:
This comment was for the wrong line. It was about:

        .doNotHaveFullyQualifiedName("jmri.jmrix.ncemonitor.NcePacketMonitorPanel").and()

Reply:
The main NCE is migrated.  The `ncemonitor` package migration is in an different PR from this one.

Reply:
Sorry, I see now that you have a new PR for this. So this is resolved.